.github/workflows/publish_docs.yaml

```
@@ -0,0 +1,43 @@
+name: 'Publish Docs'
+
+on:
+  push:
+    branches:
+      - main
```

I think the current trigger may lead to some misunderstanding. For instance, the documentation for the last version will be continually updated with any PR merged with work in progress between releases, possibly getting out of sync. I suggest using a trigger to freeze it on every release. 

```suggestion
on:
  push:
    tags:
      - '[0-9]+.[0-9]+.[0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+'
```